Remove .components-item-group selector in edit-site components[2]

im3dabasia opened the pull request:
Fixes: https://github.com/WordPress/gutenberg/issues/67369

### What
The .components-item-group selector has been replaced with a custom class (.edit-site-sidebar-navigation-screen__item-group) that is scoped specifically to the sidebar navigation screen.

This change does not affect other components outside of the sidebar navigation screen and will prevent unintended side effects if the ItemGroup component is used elsewhere in the future.

As requested in the earlier PR now closed (#67384), to apply horizontal negative margins in 4 more places ([Reference](https://github.com/WordPress/gutenberg/pull/67384#issuecomment-2507685683))

1. Navigation
2. Pages
3. Templates
4. Patterns 

### Screenshots

<img width="1469" alt="Screenshot 2024-12-04 at 6 33 57 PM" src="https://github.com/user-attachments/assets/179229f5-44ab-46dd-9bb8-1134e1850052">


#### Screenshots of the negatively applied margins.

**Navigation Screen**
![Picture 1](https://github.com/user-attachments/assets/9197107a-6827-4ed2-b647-ab985773e087)  

**Pages Screen**
![Picture 2](https://github.com/user-attachments/assets/6c1ef56c-8f2c-4155-afee-0671b866d268)  

**Templates Screen**
![Picture 3](https://github.com/user-attachments/assets/28e023d7-86c8-497d-b80b-fd7d06da827e)  

**Patterns Screen**
![Picture 4](https://github.com/user-attachments/assets/fb2c9d55-9749-447a-9ada-a82bb9ee06f2)

im3dabasia committed:
fix: Removed .components-item-group selector class usage in scss file in edit-site components

Applied horizontal margins to the following screens
1. Navigation
2. Pages
3. Templates
4. Patterns

im3dabasia committed:
fix: Added sidebar-navigation-screen-main/style.scss file correctly

im3dabasia commented:
Hi @t-hamano ,

If you have a moment, could you please review my PR? I would really appreciate your feedback.

Thank you so much!

t-hamano reviewed (commented):
Thanks for the PR! 

Sorry, maybe my explanation in #67384 was insufficient. Can you just do the following two things?

- Create a new `packages/edit-site/src/components/sidebar-navigation-screen-templates-browse/style.scss` file and move the following styles into it:
  ```scss
  .edit-site-sidebar-navigation-screen-templates-browse {
  	margin-left: -$grid-unit-20;
  	margin-right: -$grid-unit-20;
  }
  ```
- Rearrange the import statements in the `packages/edit-site/src/style.scss` file. It's a good idea to group import statements with the `sidebar-navigation-screen` prefix:
  ```scss
  @import "./components/sidebar-navigation-screen/style.scss";
  @import "./components/sidebar-navigation-screen-details-footer/style.scss";
  @import "./components/sidebar-navigation-screen-navigation-menu/style.scss";
  @import "./components/sidebar-navigation-screen-patterns/style.scss";
  @import "./components/sidebar-navigation-screen-main/style.scss";
  @import "./components/sidebar-navigation-screen-templates-browse/style.scss";
  ```

im3dabasia committed:
refactor: CSS code for sidebar in editor

im3dabasia commented:
Thank you @t-hamano , 

I’ve addressed both of the changes you requested. Please review the updated work when you have a moment.

> * Rearrange the import statements in the `packages/edit-site/src/style.scss` file. It's a good idea to group import statements with the `sidebar-navigation-screen` prefix:

There was one more file 

```css
@import "./components/sidebar-navigation-screen-navigation-menus/style.scss";
```

Since it shares the same sidebar-navigation-screen prefix, I’ve moved it accordingly as well.

Looking forward to your feedback!

t-hamano reviewed (approved):
LGTM! Thanks for working with it through to the end 👍

Note: Native E2E test failures are not related to this PR. See https://github.com/WordPress/gutenberg/pull/67595

afercia commented:
@im3dabasia @t-hamano looks like the ItemGroup in the first level of the navigation panel isn't aligned correctly and misses the negative margins. Can you please have a look at it? Screenshot from latest trunk:

![Screenshot 2024-12-11 at 09 54 48](https://github.com/user-attachments/assets/d9df71aa-6f0d-48e7-ae63-7c6dbd7128c3)

im3dabasia commented:
Hi @afercia,

I checked the latest trunk, and the negative margins are applied correctly on my end. I’ve attached a screenshot for reference.

<img width="1468" alt="image" src="https://github.com/user-attachments/assets/afcf67ef-f2ea-4652-8bb0-7324d8c21872">

afercia commented:
@im3dabasia based on your screenshot, the negative margins are applied to the `.components-item-group` selector. That is not used any longer on latest trunk. Avoiding the usage of `components-item-group` was the whole point of this PR so I'm not sure you're looking at the actual latest trunk.

im3dabasia commented:
Yeah @afercia , I had the latest trunk version, I restarted the repo and then checked and yes what you claim holds true. The negative margins are not applied in the 'Design' menu. 

I'll work on this right away.

t-hamano commented:
Thanks for noticing. This issue is a regression introduced by #66851. Just adding the `edit-site-sidebar-navigation-screen-main` CSS class [here](https://github.com/WordPress/gutenberg/blob/b65823092b1a319ac533be48d33a09f5f9a239d1/packages/edit-site/src/components/sidebar-navigation-screen-main/index.js#L22) should solve it.

@im3dabasia I'll be happy to review your PR.

im3dabasia commented:
@afercia,

It looks like the issue occurred due to this [commit](https://github.com/im3dabasia/gutenberg/commit/7159a7857fb9106c4ecaa299220de91f0fff23c8) which was merged recently, where the `className="edit-site-sidebar-navigation-screen-main"` I introduced in <ItemGroup> was removed. This seems to have caused the negative margins not to apply.

Here's the relevant [diff](https://github.com/im3dabasia/gutenberg/commit/7159a7857fb9106c4ecaa299220de91f0fff23c8#diff-ecffee0daf6148e5e4693a18d52efb690aeb1db805f961e0466114da832ae6b6L21).

Would it be helpful if I opened a new PR to reintroduce the className, or is there a different approach you would prefer to resolve this?

im3dabasia commented:
> @im3dabasia I'll be happy to review your PR.

Sure @t-hamano , I'll raise a new PR and cc you ASAP.

afercia commented:
> It looks like the issue occurred due to this [commit](https://github.com/im3dabasia/gutenberg/commit/7159a7857fb9106c4ecaa299220de91f0fff23c8) which was merged recently,

Yes thanks, I do see that commit removed the newsly added class. Please do feel free to raise a quick PR to re-add it and kindly reference this PR and https://github.com/WordPress/gutenberg/pull/66851 in your new PR, thanks!
